### PR TITLE
Add eventhubs tasks

### PR DESCRIPTION
**Pull request by Blackbaud-AlexFortin**
@blackbaud/highlander @Blackbaud-MikeLueders

### CODE REVIEW

File: src/main/groovy/com/blackbaud/templates/tasks/EventHubsProject.groovy [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+eventhubs.consumer.storageAccountName=eventhubcommits
+eventhubs.consumer.storageAccountKey=anotherFakeKey
+eventhubs.consumer.storageAccountContainer=testhub-checkpoints
+
+eventhubs.stub=true
+""")
+    }
+
+    void addApiObject(String resourceName) {
+        basicProject.addExternalApiObject("eventhubs", resourceName, false)
```
Comment (Blackbaud-MikeLueders) [inline src/main/groovy/com/blackbaud/templates/tasks/EventHubsProject.groovy]: consider adding support for 'internal' flag - see the service bus stuff for an example

Comment (Blackbaud-MikeLueders) [reply src/main/groovy/com/blackbaud/templates/tasks/EventHubsProject.groovy]: in summary, it's just supporting an `internal` flag from the command line and then selectively calling `addExternalApiObject`/`addInternalApiObject`

Comment (Blackbaud-AlexFortin) [reply src/main/groovy/com/blackbaud/templates/tasks/EventHubsProject.groovy]: 👍

**Comment by MatthewMazaika** [comment]
:clap:

File: src/main/groovy/com/blackbaud/templates/tasks/AddEventHubsContainerTask.groovy [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.blackbaud.templates.tasks
+
+import org.gradle.api.tasks.TaskAction
+
+
+class AddEventHubsContainerTask extends AbstractTemplateTask {
+
+    AddEventHubsContainerTask() {
+        super("Add a eventhubs container and default configuration to an existing project (options: -Pname=?)")
```
Comment (Blackbaud-BethRichardson) [inline src/main/groovy/com/blackbaud/templates/tasks/AddEventHubsContainerTask.groovy]: Two places: "add an eventhubs" (typo)

File: src/main/groovy/com/blackbaud/templates/tasks/AddEventHubsMessageTask.groovy [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package com.blackbaud.templates.tasks
+
+import org.gradle.api.tasks.TaskAction
+
+class AddEventHubsMessageTask extends AbstractTemplateTask {
+
+    AddEventHubsMessageTask() {
+        super("Adds a eventhubs message and random builder skeleton (options: -Pname=?, -Pinternal)")
```
Comment (Blackbaud-BethRichardson) [inline src/main/groovy/com/blackbaud/templates/tasks/AddEventHubsMessageTask.groovy]: other place

File: src/main/groovy/com/blackbaud/templates/tasks/AddEventHubsContainerTask.groovy (hunk 2 above)
Comment (Blackbaud-BethRichardson) [reply src/main/groovy/com/blackbaud/templates/tasks/AddEventHubsContainerTask.groovy]: Might want to remove mention of container and focus on adding a hub

File: src/main/resources/templates/springboot/eventhubs/build.eventhubs-client.gradle.tmpl [hunk 4]
```diff
@@ -0,0 +1,9 @@
+ext.artifactId = "\${artifactId}-eventhubs-client"
+
+dependencies {
+    compile "com.blackbaud:common-eventhubs:1.+"
```
Comment (Blackbaud-BethRichardson) [inline src/main/resources/templates/springboot/eventhubs/build.eventhubs-client.gradle.tmpl:4]: Should this use a parameter for "commonEventHubsVersion"

File: src/main/resources/templates/springboot/eventhubs/resource-api.java.tmpl [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class ${resourceName} {
```
Comment (Blackbaud-BethRichardson) [inline src/main/resources/templates/springboot/eventhubs/resource-api.java.tmpl:12]: should this be called something other than "resource"?

**Review by Blackbaud-BethRichardson — DISMISSED** [review]

File: src/main/resources/templates/springboot/eventhubs/build.eventhubs-client.gradle.tmpl (hunk 4 above)
Comment (Blackbaud-AlexFortin) [reply src/main/resources/templates/springboot/eventhubs/build.eventhubs-client.gradle.tmpl:4]: This is a separate build.gradle only for the client so this is the only place the version will be mentioned so its probably overkill. If you need to change it you just change it in this one spot

File: src/main/resources/templates/springboot/eventhubs/resource-api.java.tmpl (hunk 5 above)
Comment (Blackbaud-AlexFortin) [reply src/main/resources/templates/springboot/eventhubs/resource-api.java.tmpl:12]: think its fine. If hey really want to change the name they can just refactor it. This is just laying down the foundation/structure

**Pull request merged**